ple/games/donkeykong/Board.py

```diff
@@ -0,0 +1,335 @@
+__author__ = 'Batchu Vishal'
+import pygame
+import math
+import sys
+
+from Person import Person
+from OnBoard import OnBoard
+from Coin import Coin
+from Player import Player
+from Fireball import Fireball
+from Donkey import Donkey
+
+'''
+This class defines our gameboard.
+A gameboard contains everthing related to our game on it like our characters, walls, ladders, coins etc
+The generation of the level also happens in this class.
+'''
+
+
+class Board:
+    def __init__(self, width, height, rewards, rng):
+        self.__width = width
+        self.__actHeight = height
+        self.__height = self.__actHeight + 10
+        self.score = 0
+        self.rng = rng
+        self.rewards = rewards
+        self.gameState = 1
+        self.cycles = 0  # For the characters animation
+        self.direction = 0
+
+        self.white = (255, 255, 255)
+
+        '''
+        The map is essentially an array of 30x80 in which we store what each block on our map is.
+        1 represents a wall, 2 for a ladder and 3 for a coin.
+        '''
+        self.map = []
+        # These are the arrays in which we store our instances of different classes
+        self.Players = self.Enemies = self.Allies = self.Coins = self.Walls = self.Ladders = self.Fireballs = self.Hearts = self.Boards = self.FireballEndpoints = []
+
+        # Resets the above groups and initializes the game for us
+        self.resetGroups()
+
+        self.myfont = pygame.font.SysFont("comicsansms", 50)
+
+        self.background = pygame.image.load('Assets/background.png')
+        self.background = pygame.transform.scale(self.background, (width, height))
```

Comment:
performance: might be useful to use image.convert() or image.convert_alpha() depending on if the image uses alpha